src/gsMpi/gsMpiComm.h

```diff
@@ -141,6 +141,36 @@ class gsSerialComm
         return 0;
     }
 
+    /** @brief Sends data to a destination process with a defined tag (blocking)
+     *
+     * One process sends data of length len to the destination process dest. The argument tag specifies the message ID.
+     *
+     * @param[in] in The send buffer with the data to send
+     * @param[in] len The number of elements which will be sent
+     * @param[in] dest The rank of the process which should receive the message
+     * @param[in] tag Specifies the message ID
+     */
+    template<typename T>
+    static int send (T* in, int len, int dest, int tag = 0)
+    {
+        return 0;
+    }
+
+    /** @brief Receives data from a source process with a defined tag (blocking)
+     *
+     * One process receives data of length len from the source process source. The argument tag specifies the message ID.
+     *
+     * @param[out] out The buffer to store the received data in
+     * @param[in] len The number of elements which will be received
+     * @param[in] source The rank of the process which sended the message
+     * @param[in] tag Specifies the message ID
+     */
+    template<typename T>
+    static int recv (T* out, int len, int source, int tag = 0)
+    {
+        return 0;
+    }
+
```

Comment:
I think Isend and Irecv should also be added here

Reply:
For Isend and Irecv I need additionally a MPI_Request structure as an argument.

Reply:
OK, for now just add
typedef int MPI_Request; 
in gsSerialComm.

This allows to add the functions in gsSerialComm as well